tests/python/pants_test/reporting/test_reporting.py

```diff
@@ -178,3 +187,28 @@ def test_raise_if_trace_id_is_of_wrong_ch_format(self):
       + "Got {}.".format(trace_id)
       in str(result.exception)
     )
+
+  def test_poll(self):
+    with temporary_dir() as dir:
+      class TestPantsHandler(PantsHandler):
+        def __init__(self, request, client_address, server):
+          super(TestPantsHandler, self).__init__(
+            settings=ReportingServer.Settings(
+              info_dir=dir,
+              template_dir=dir,
+              assets_dir=dir,
+              root=dir,
+              allowed_clients=['ALL'],
+            ),
+            renderer=None,
+            request=request,
+            client_address=client_address,
+            server=server,
+          )
+      safe_file_dump(os.path.join(dir, "file"), "hello")
+      with http_server(TestPantsHandler) as port:
+        response = requests.get("http://127.0.0.1:{}/poll?{}".format(
+          port,
+          urlencode({"q": json.dumps([{"id": "0", "path": "file"}])}),
+        ))
+        self.assertEqual(response.json(), {"0": "hello"})
```

Comment:
Nit: unindent this as it no longer needs to be a part of the `with` statement.

Reply:
Done